### PR TITLE
Add exported path to be able to export the ca private key

### PR DESCRIPTION
**Pull request by QuentinBisson**

### CODE REVIEW

File: create.go [hunk 1]
```diff
@@ -27,17 +27,56 @@ func (p *VaultPKI) CreateBackend(ID string) error {
...
+		PrivateKey:  privateKey,
+	}, nil
+}
+
+func (p *VaultPKI) CreateCA(ID string) (CertificateAuthority, error) {
+	return p.createNewCA(ID, true)
+}
+
+func (p *VaultPKI) CreateCAWithPrivateKey(ID string) (CertificateAuthority, error) {
+	return p.createNewCA(ID, true)
```
Comment (stone-z) [inline create.go]: I suspect one of these should be `false`

**Review by stone-z — APPROVED** [review]

File: create.go (hunk 1 above)
Comment (QuentinBisson) [reply create.go]: Thanks for noticing :D

**Comment by xh3b4sd** [comment]
We have some breaking changes here. The tag was a patch release. Somebody working in that area runs into errors thrown by the compiler left and right. There was no communication about this. I do not see that anyone is eagerly working on a fix in `cert-operator`. Here are also a couple of questionable design decisions from a software point of view and what you plan to do with exposing the private keys of the root CA doesn't sound super solid from a security point of view. I feel like a couple of things go the wrong way here.

**Comment by QuentinBisson** [comment]
Feel free to revert the change. This commit was made to create a POC for the cert rotation story which was put on hold due to spec discussions where we actually need the private key to allow the controller manager to sign CSR requests. If you think we could do that another way, do not hesitate to take part in the spec discussion :p

Regarding the breaking changes, this is the reason I went with a new tag so the issue could be that the project is using master as a dependency instead of a tagged version. I suppose I should have crafted a 0.2.0 release though

Regarding communication, is there a doc I can follow to know what should be communicated? As it is my first code PR, I do not really not what I should be doing.

**Comment by xh3b4sd** [comment]
We should try to adhere to semver. Breaking changes mean a major version bump. See https://semver.org. So `v0.2.0` for this here would also not have been sufficient. I am not sure anything is written down but you would be good to go with a message in `#news-dev` to create awareness. I checked the spec and saw that a lot of competent people did already challenge the approach so I did not see a personal need to intervene there as well.

**Comment by QuentinBisson** [comment]
Oh true, my bad for not following semver there :( Do not hesitate to remove the commit as it seems the spec tends to go to generating the private key for the intermediate CA. I will create version a major version at a later date

Thanks for the info on awareness btw

**Pull request merged**